c7n/resources/ecs.py

```diff
@@ -594,26 +594,48 @@ class DeleteTaskDefinition(BaseAction):
     The definition will be marked as InActive. Currently running
     services and task can still reference, new services & tasks
     can't.
+    
+    deregister is True by default. When given as False, the task definition will 
+    be permanently deleted.
+    
+    .. code-block:: yaml
+
+       policies:
+         - name: deregister-task-definition
+           resource: ecs-task-definition
+           filters:
+             - family: test-task-def
+           actions:
+             - type: delete
+         - name: delete-task-definition
+           resource: ecs-task-definition
+           filters:
+             - family: test-task-def
+           actions:
+             - type: delete
+               deregister: False
     """
 
-    schema = type_schema('delete')
-    permissions = ('ecs:DeregisterTaskDefinition',)
+    schema = type_schema('delete', deregister={'type': 'boolean'})
```

Comment:
imo, we should pick up a better name here. `deregister` may confuse people since this is also an action type for ECS. Thoughts @ajkerrigan ?

Reply:
True, would `permanently` be better? (defaulted to false)

Reply:
Damn naming :sweat_smile: . So roundup of options from this thread...

- `deregister`: potentially confusing, fair point
- `deregister-only`: a bit more explicit, but also longer and a bit awkward
- `permanently`: clearer and shorter, but from a schema perspective we're "inventing a word" with no ties to the ECS service api or other custodian actions
- `force`: consistent with some other custodian actions, but also potentially confusing

Regardless of the word we choose, looks like we're all on board with keeping the default deregister-only _behavior_ and allowing this opt-in delete.

Fair summary? If so, I _think_ I'm leaning toward `force` or `deregister-only`, but it's a pretty weak leaning. I think we can make any of those options work with an example in the action docs.